skimage/filters/thresholding.py

```diff
@@ -580,9 +596,26 @@ def threshold_li(image, *, tolerance=None):
     image -= image_min
     tolerance = tolerance or np.min(np.diff(np.unique(image))) / 2
 
-    # Initial estimate
-    t_curr = np.mean(image)
-    t_next = t_curr + 2 * tolerance
+    # Initial estimate for iteration. See "initial_guess" in the parameter list
+    if initial_guess is None:
+        t_next = np.mean(image)
+    elif callable(initial_guess):
+        t_next = initial_guess(image)
+    elif np.isscalar(initial_guess):  # convert to new, positive image range
+        t_next = initial_guess - image_min
```

Comment:
What happens if this is negative?

Reply:
Nothing good! Actually any value outside of the range of the initial data will cause the loop to fail. I'll add a check for that!